pkg/gofr/datasource/redis/redis.go

```diff
@@ -33,10 +37,11 @@ type Redis struct {
 	config *Config
 }
 
-// NewClient return a redis client if connection is successful based on Config.
+// NewClient returns a redis client if connection is successful based on Config.
+// Supports both plain and TLS connections. TLS is configured via REDIS_TLS_ENABLED and related environment variables.
 // In case of error, it returns an error as second parameter.
```

Comment:
There is something strange here. I'm unsure there is something we can do.

`redis` package has a `Config` struct, NewClient uses the `Config` struct of `config` package.

The go doc here talks about 

> NewClient returns a redis client (…) based on Config

So it brings confusion.

So maybe this would help

```suggestion
// NewClient returns a [Redis] client if connection is successful based on [config.Config].
// Supports both plain and TLS connections. TLS is configured via REDIS_TLS_ENABLED and related environment variables.
// In case of error, it returns an error as second parameter.
```